test/e2e/tests/network/update-network.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { strict as assert } from 'assert';
+import { WebElement } from 'selenium-webdriver';
+import { Suite } from 'mocha';
+import FixtureBuilder from '../../fixture-builder';
+import {
+  defaultGanacheOptions,
+  unlockWallet,
+  withFixtures,
+} from '../../helpers';
+import { Driver } from '../../webdriver/driver';
+
+const selectors = {
+  accountOptionsMenuButton: '[data-testid="account-options-menu-button"]',
+  informationSymbol: '[data-testid="info-tooltip"]',
+  inputText: 'input[type="text"]',
+  settingsOption: { text: 'Settings', tag: 'div' },
+  networkOption: { text: 'Networks', tag: 'div' },
+  ethereumNetwork: { text: 'Ethereum Mainnet', tag: 'div' },
+  deleteButton: { text: 'Delete', tag: 'button' },
+  cancelButton: { text: 'Cancel', tag: 'button' },
+  saveButton: { text: 'Save', tag: 'button' },
+  updatedNetworkDropDown: { tag: 'span', text: 'Update Network' },
+  errorMessageInvalidUrl: {
+    tag: 'h6',
+    text: 'URLs require the appropriate HTTP/HTTPS prefix.',
+  },
+};
+
+async function editNetworkDetails(
+  driver: Driver,
+  indexOfInputField: number,
+  inputValue: string,
+) {
+  const getAllInputElements: WebElement[] = (await driver.findElements(
+    selectors.inputText,
+  )) as WebElement[];
+  const inputTextFieldToEdit: WebElement =
+    getAllInputElements[indexOfInputField];
+  await inputTextFieldToEdit.clear();
+  await inputTextFieldToEdit.sendKeys(inputValue);
+}
+
+async function navigateToEditNetwork(driver: Driver) {
+  await driver.clickElement(selectors.accountOptionsMenuButton);
+  await driver.clickElement(selectors.settingsOption);
+  await driver.clickElement(selectors.networkOption);
+}
+describe('Update Network:', function (this: Suite) {
+  it('name is updated and validate the ui elements', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test?.fullTitle(),
+      },
+
+      async ({ driver }: { driver: Driver }) => {
+        await unlockWallet(driver);
+        await navigateToEditNetwork(driver);
+        await editNetworkDetails(driver, 2, 'Update Network');
+        await driver.clickElement(selectors.saveButton);
+
+        // Validate the network name is updated
+        const networkName = await driver.findElement(
```

Comment:
This is a case where we should include the expected text as part of the `findElement` call, instead of getting the element first and then getting the text later.

The (soon to be merged) [e2e test contributor documentation](https://github.com/MetaMask/contributor-docs/pull/72/files?short_path=0444184#diff-0444184f97e15aca1b7c0cf9256541e639367d3d479d80d1277d2151c72785b7) provides an example of this:
```
const isExpectedBoxContentPresentAndVisible =
  await driver.isElementPresentAndVisible({
    css: '[data-testid="mm-banner-alert-notification-text"]',
    text: options.text,
  });
assert.equal(
  isExpectedBoxContentPresentAndVisible,
  true,
  'Invalid box text content',
);
```